File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/ProductRef.java
package com.sdl.ecommerce.api.model;

/**
 * ProductRef
 *
 * @author nic
 */
public class ProductRef {

    private String id;
    private String name;

    public ProductRef(Product product) {
        this.id = product.getId();
        this.name = product.getName();
    }

    public ProductRef(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.Rest/Model/ProductRef.cs
using SDL.ECommerce.Api;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SDL.ECommerce.Rest.Model
{
    public class ProductRef : IProductRef
    {
        public string Id { get; set; }
        public string Name { get; set; }
    }
}
